### ISSUE TITLE
Коментар до 3 завдання

### ISSUE THREAD

**Issue by akhsunna** [Nasya/RentSkiingGear#2]
Щодо перших двох пунктів
Замініть ваш тест в моделі на цей:

```
  context 'when create new user' do
    let(:city) { City.create(name: 'Kyiv', id: 4) }

    it 'shows it in the users list for city' do
      user = city.users.create(name: 'Mike')
      expect(city.users).to eq([user])
      expect(user).to be_persisted
    end

    it 'shows items of the user in the items list for city' do
      user = city.users.create(name: 'Mike')
      5.times { user.items.create(name: 'some_item') }

      expect(city.items).to eq(user.items)
    end
  end
```

**Comment by Nasya** [comment]
Падає на 2-му тесті:
![2018-03-14 12-51-35](https://user-images.githubusercontent.com/8141921/37398215-72eb7e0e-2786-11e8-92c1-123cbccf79a9.png)

**Comment by akhsunna** [comment]
Ну але він на вашу думку правильний чи ні?

**Comment by Nasya** [comment]
ми створюємо юзера в нашому місті.
Пізніше цей юзер створює в циклі 5 предметів.

Далі перевірка чи предмети з нашого міста відповідаю предметам нашого юзера.
В результаті якої предмети в місті відображаються, а предмети юзера пусті.
Чому так я, насправді, поки не дуже розумію.

**Comment by Nasya** [comment]
замінила асоціацію на:
`has_many :items, through: :users`
і тест пройшов.
Це правильне рішення?

**Comment by akhsunna** [comment]
Ні, навпаки, пусті якраз предмети для міста, а для юзера все гаразд.
бо ми `expect`, що `city.items` буде рівне `user.items`, але воно пусте

Так, правильне
Але важливо зрозуміти чому попередній варіант звязку неправильний і як воно працювало

**Comment by Nasya** [comment]
Не в ту сторону `expect` прочитала.

Я шукала items через зовнішній ключ user_id  в моделі Item. Я правильно трактую?
Поки не дуже можу уявити як воно працювало.

П.С. трохи не комфортно писати, що щось не знаю чи не розумію, але як на мене краще спитати, аніж вдавати, що все ясно, а потім ті костилі десь тай вилізуть ))

**Comment by akhsunna** [comment]
Так, воно шукало в таблиці items записи, в яких user_id рівне айдішці міста.

Може буде наглядніше якщо запустете консоль з попереднім варіантом (`has_many :items, foreign_key: :user_id, class_name: 'Item', dependent: :destroy`) 
і спробуєте подіставати items для city. тоді одразу буде вийдно sql-запити

**Comment by Nasya** [comment]
Дякую)

Питання щодо 3 завдання:
> Provide a way to fetch all reviews for all items of particular user.

В мене була думка, що це реалізується через join table migration, але тепер я не впевнена.
Чи це все ж реалізується через асоціації?
Але ж Reviewable реалізований через concern.

**Comment by akhsunna** [comment]
Все ще через асоціації

А які проблеми з concern?

**Comment by Nasya** [comment]
Не те сказала.

Пробую ось таку асоціацію в моделі **Item**.
```
  has_many :reviewed_items, through: :reviews,
           source: :reviewable, source_type: 'Item'
```
Або я не то пробую, або в тесті єрєсь пишу.

**Comment by akhsunna** [comment]
чому `through: :reviews`? 
`.reviews` -- це ж відгуки, залишені людині, а не речі

**Comment by Nasya** [comment]
ми раніше в тестах викликали reviews для item:
 ```
   it 'shows it in the received reviews list for item' do
      item_review = author.authored_reviews.create(reviewable: item, rw_text: 'Best board')
      expect(item.reviews).to eq([item_review])
      expect(item_review).to be_persisted
    end
```

**Comment by akhsunna** [comment]
Перепрошую, упустила, що ви це робите в моделі **Item**

**Comment by akhsunna** [comment]
користувач має багато речей
речі мають багато отриманих відгуків
нам треба, щоб можна було гарно і легко дістати всі відгуки для речей користувача

**Comment by akhsunna** [comment]
ви вже зробили аналогічне для міста
місто має багато користувачів
користувач має багато речей
і ми для міста дістаємо речі
через користувачів
тут той самий принцип

**Comment by Nasya** [comment]
Зараз спробую)

**Comment by Nasya** [comment]
Попробувала:
`  has_many :items_reviews, through: :items, source: :reviews`
в User моделі
по попередньому прикладу, але знову ж таки не впевнена чи не правильно асоціація чи тести не правильні написала.

**Comment by akhsunna** [comment]
Як на мене, то код правильний, мав би працювати

**Comment by Nasya** [comment]
Значить тест не правильний)

**Comment by Nasya** [comment]
```
    it 'shows it in the received reviews list for item' do
      5.times { user.authored_reviews.create(reviewable: item, rw_text: 'Best board') }
      expect(user.items_reviews).to eq(item.reviews)
    end
```
такий тест зробила.
